src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/SignedXml.cs

```diff
@@ -354,14 +354,21 @@ public bool CheckSignature(X509Certificate2 certificate, bool verifySignatureOnl
 
                 // Do the chain verification to make sure the certificate is valid.
                 X509Chain chain = new X509Chain();
-                chain.ChainPolicy.ExtraStore.AddRange(BuildBagOfCerts());
-                bool chainVerified = chain.Build(certificate);
-                SignedXmlDebugLog.LogVerifyX509Chain(this, chain, certificate);
-
-                if (!chainVerified)
+                try
                 {
-                    SignedXmlDebugLog.LogVerificationFailure(this, SR.Log_VerificationFailed_X509Chain);
-                    return false;
+                    chain.ChainPolicy.ExtraStore.AddRange(BuildBagOfCerts());
+                    bool chainVerified = chain.Build(certificate);
+                    SignedXmlDebugLog.LogVerifyX509Chain(this, chain, certificate);
+                
+                    if (!chainVerified)
+                    {
+                        SignedXmlDebugLog.LogVerificationFailure(this, SR.Log_VerificationFailed_X509Chain);
+                        return false;
+                    }
+                }
+                finally
+                {
+                    chain.Dispose();
```

Comment:
If only the variable is disposed and nothing else, then `using` should be better.